### ISSUE TITLE
Missing https://docs.rs/embedded-hal/1.0.0/embedded_hal/spi/trait.SpiDevice.html impl

### ISSUE THREAD

**Issue by bjoernQ** [esp-rs/esp-hal#2778]

**Comment by jessebraham** [comment]
Adding a new third-party trait implementation shouldn't be a breaking change, should it? Don't recall why we decided to implement this, it should be additive functionality AFAIK. Am I missing/forgetting something?

**Comment by bugadani** [comment]
I think this is in "investigation" because I'm stubbornly keep repeating that the SPI bus drivers we have should not implement SpiDevice anyway, and that deserved the investigation label instead of a 10-minute detour.

**Comment by bugadani** [comment]
FWIW the resolution of this issue is still just adding more code so the 1.0-blocker can probably be removed.

**Comment by MabezDev** [comment]
Agree, we can add this post 1.0 as it will be a completely new set of structs.
